src/promise.js

```diff
@@ -749,12 +756,21 @@ function deferResolve(v) {this.promise._resolveCallback(v);}
 function deferReject(v) {this.promise._rejectCallback(v, false);}
 
 Promise.defer = Promise.pending = function() {
-    debug.deprecated("Promise.defer", "new Promise");
+    // TODO HAD-15960 remove shim for deprecated APIs,
+    //      and update callers  to use recommended API
+    // will fix the defer instances with new promise later.
+    // debug.deprecated("Promise.defer", "new Promise");
     var promise = new Promise(INTERNAL);
     return {
         promise: promise,
         resolve: deferResolve,
-        reject: deferReject
+        reject: deferReject,
+        // TODO HAD-15960 remove shim for deprecated APIs,
+        //      and update callers  to use recommended API
+        fulfill: deferResolve,
+        cancel: function () {
+            return this.promise.cancel();
+        }
```

Comment:
Why was promise.cancel() removed? That seems like a step backward in usability to me. What do you do now if you want to cancel a promise and you don't have access to its resolver? Or are you not able to do that now (and maybe we don't ever need to?)?